lib/galaxy/config.py

```diff
@@ -363,9 +365,9 @@ def __init__( self, **kwargs ):
             # so dependending on the context a better default can be used (
             # request url in a web thread, Docker parent in IE stuff, etc...)
             galaxy_infrastructure_url = "http://localhost"
-            port = self.guess_galaxy_port()
-            if port:
-                galaxy_infrastructure_url += ":%s" % (port)
+            web_port = self.galaxy_infrastructure_web_port or self.guess_galaxy_port()
+            if web_port:
+                galaxy_infrastructure_url += ":%s" % (web_port)
```

Comment:
``guess_galaxy_port()`` is correctly figuring out that the Galaxy process is being hosted on 8080 - but with ``REMOTE_USER`` enable you cannot directly use the API on port 8080 (that request will not contain a REMOTE_USER and so will be rejected). I think anyway...